sql/parser/builtins.go

```diff
@@ -66,9 +67,16 @@ func (b builtin) match(types typeList) bool {
 // sorted please.
 var builtins = map[string][]builtin{
 	// TODO(XisiHuang): support encoding, i.e., length(str, encoding).
-	"length": {stringBuiltin1(func(s string) (Datum, error) {
-		return DInt(len(s)), nil
-	}, DummyInt)},
+	"length": {
+		stringBuiltin1(func(s string) (Datum, error) {
+			return DInt(utf8.RuneCountInString(s)), nil
+		}, DummyInt),
+		bytesBuiltin1(func(s string) (Datum, error) {
+			return DInt(len(s)), nil
+		}, DummyInt),
+	},
+
+	// TODO(pmattis): What string functions should also support bytesType?
```

Comment:
`to_hex` takes an int parameter. I'm not sure why `sha256` (and `sha1` and `md5`) should be restricted to bytes. Those functions definitely make sense on strings.